shared/src/main/scala/coop/rchain/sdk/dag/data/DagView.scala

```diff
@@ -0,0 +1,25 @@
+package coop.rchain.sdk.dag.data
+
+import fs2.{Chunk, Stream}
+
+/**
+  * Represents read-only view of the DAG starting from latest messages.
+  */
+trait DagView[F[_], M, S] {
+  implicit val dd: DagData[M, S]
+  implicit val ordM: Ordering[M]
```

Comment:
This is important part of the abstraction. It gives additional dependency to interface that can be used to implement domain logic directly on the interface and not on the implementation (like we have on MultiParentCasper).

Implementation should be straightforward without domain logic and should only be connection to underlying side-effects like DB, FS or network operations.